File: content/code-examples/tested/javascript/driver/aggregation/pipelines/join-one-to-one/tutorial.snippet.set.js
pipeline.push(
  {
    $set: {
      product_mapping: { $first: '$product_mapping' },
    },
  },
  {
    $set: {
      product_name: '$product_mapping.name',
      product_category: '$product_mapping.category',
    },
  }
);

File: content/code-examples/tested/java/driver-sync/aggregation/pipelines/join1to1/Tutorial.snippet.set.java
pipeline.add(Aggregates.set(
        new Field<>(
                "product_mapping",
                new Document("$first", "$product_mapping")
        )
));

pipeline.add(Aggregates.set(
        new Field<>("product_name", "$product_mapping.name"),
        new Field<>("product_category", "$product_mapping.category")
));
